storage/src/system/protocol_upgrade.rs

```diff
@@ -1132,6 +1133,38 @@ where
         Ok(())
     }
 
+    /// Handle unbonds migration.
+    pub fn handle_unbonds_migration(&mut self) -> Result<(), ProtocolUpgradeError> {
+        debug!("handle unbonds migration");
+        let tc = &mut self.tracking_copy;
+        let existing_keys = match tc.get_keys(&KeyTag::Unbond) {
+            Ok(keys) => keys,
+            Err(err) => return Err(ProtocolUpgradeError::TrackingCopy(err)),
+        };
+        for key in existing_keys {
+            if let Some(StoredValue::Unbonding(unbonding_purses)) =
+                tc.get(&key).map_err(Into::<ProtocolUpgradeError>::into)?
+            {
+                // prune away the original record, we don't need it anymore
```

Comment:
Should we have an if condition to only migrate unbonding purses based on the unbonding delay, i.e what we did for redelegate, only migrate the purses that are yet to be paid

Reply:
Maybe. However, the original logic was a stored value of a vec of unbonding purses which were being popped off for a pseudo prune so there shouldn't be any such records.

On that note tho, an optimization to not write the record if unbond.eras().len() == 0 would certainly be feasible.